packages/rscode/src/extension.ts

```diff
@@ -179,6 +181,80 @@ ${text}
     }
   );
 
+  // Add Docs command
+  const addDocs = vscode.commands.registerTextEditorCommand(
+    'extension.rocksetAdd',
+    async (activeEditor) => {
+      try { // try to parse JSON
+        const docs = JSON.parse(activeEditor.document.getText());
+        try {
+          client.workspaces.listWorkspaces().then(function (rawWorkspaces) { // list workspaces
+            let workspaces: string[] = []
+
+            rawWorkspaces.data?.forEach(item => { // for each workspace, add it's name to `workspaces`
+              if (typeof (item.name) === "string") { workspaces.push(item.name) }
+            });
+            vscode.window.showQuickPick(workspaces, { placeHolder: "workspace" }).then(workspace => { // show dropdown menu of workspaces
+              if (!workspace) { return } // if user exits, return
+
+              client.collections.workspaceCollections(workspace).then(function (rawCollections) { // list collections in workspace
+                let collections: string[] = []
+                rawCollections.data?.forEach(item => {
+                  if (typeof (item.name) === "string") { collections.push(item.name) }
+                });
+                vscode.window.showQuickPick(collections, { placeHolder: "collection" }).then(collection => {
+                  if (!collection) { return }
+
+                  client.documents.addDocuments(workspace, collection, { // add documents
+                    data: Array.isArray(docs) ? docs : [docs] // if it is a single document, wrap it in a list
+                  }).then(
+                    function () { vscode.window.showInformationMessage("Document added.") } // show info message
+                  ).catch(vscode.window.showErrorMessage)
+                })
+              })
+            });
+          }).catch(vscode.window.showErrorMessage)
+        } catch (err) { await vscode.window.showErrorMessage(err.message) }
+      }
+      catch (err) {
+        if (err.name === "SyntaxError") {
+          // JSON is invalid, show error
+          await vscode.window.showErrorMessage("Invalid JSON  document body. See https://docs.rockset.com/rest-api/#adddocuments.");
```

Comment:
what is the response from the API in this case? why not just show err.message always?

Reply:
I have put this in so that if the document cannot be parsed, the error is immediately shown, without needing to make an API request. Is it a better solution to show the API error?

Reply:
ah I see, this is catching JSON parse errors. that makes sense. thanks for clarifying.